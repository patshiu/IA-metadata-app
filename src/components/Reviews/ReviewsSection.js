import React from 'react';
import PropTypes from 'prop-types';

import SectionTitle from '../SectionTitle/SectionTitle';
import ReviewItem from './ReviewItem';

const ReviewsSection = ({ reviews }) => {
  if (!reviews || reviews.length === 0) {
    return (
      <div className="row">
        <div className="section item-reviews">
          <SectionTitle title="Reviews" count={null} />
          <p>This item has no reviews.</p>
        </div>
      </div>
    );
  } else {
    return (
      <div className="row">
        <div className="section item-reviews">
          <SectionTitle title="Reviews" count={Object.keys(reviews).length} />
          {reviews.map((review, index) => (
            <ReviewItem key={review.review_id ? review.review_id : index} review={review} />
          ))}
        </div>
      </div>
    );
  }
};

ReviewsSection.propTypes = {
  reviews: PropTypes.array
};

export default ReviewsSection;
