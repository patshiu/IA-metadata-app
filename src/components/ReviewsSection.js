import React from 'react';
import PropTypes from 'prop-types';

import ReviewItem from './ReviewItem';

const ReviewsSection = ({ reviews }) => {
  if (!reviews) {
    return (
      <div className="section item-reviews">
        <h3>Reviews</h3>
        <p>This item has no reviews.</p>
      </div>
    );
  } else {
    return (
      <div className="section item-reviews">
        <h3>Reviews ({Object.keys(reviews).length})</h3>
        {reviews.map((review, index) => (
          <ReviewItem key={review.review_id ? review.review_id : index} review={review} />
        ))}
      </div>
    );
  }
};

ReviewsSection.propTypes = {
  reviews: PropTypes.array
};

export default ReviewsSection;
