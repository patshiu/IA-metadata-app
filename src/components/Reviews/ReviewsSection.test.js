import React from 'react';
import renderer from 'react-test-renderer';

import ReviewsSection from './ReviewsSection';

it('renders review section with no reviews correctly', () => {
  const reviews = [];
  const tree = renderer.create(<ReviewsSection reviews={reviews} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders review section with undefined reviews object correctly', () => {
  const reviews = undefined;
  const tree = renderer.create(<ReviewsSection reviews={reviews} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders review section with several reviews correctly', () => {
  const reviews = [
    {
      review_id: '6198',
      reviewbody:
        "Wonderfully trippy little cartoon on the function of computers. The color's a little muddy, but it explains the use of computers--in the service of man, not its master--as a tool for the control of information and work, design, and for simulating reality.\n\nNot sure what need this short really fulfills, but it's fun (kinda like a Fractured Fairytale-style), doesn't pander, and doesn't panic.",
      reviewtitle: 'If I only had a heart',
      reviewer: 'dynayellow',
      reviewdate: '2003-09-15 15:54:20',
      createdate: '2003-09-15 15:54:20',
      stars: '3'
    },
    {
      review_id: '6536',
      reviewbody:
        "This short film was written, produced and directed by Charles and Rae Eames for the IBM Pavillion at the 1958 Brussels World's Fair. Animation by John Whitney. Music by Elmer Bernstein. Topic is the computer in the context of human development.  MUST VIEW!",
      reviewtitle: 'A gem from the Late 50s - Charles and Rae Eames',
      reviewer: 'Steve R',
      reviewdate: '2003-10-02 01:25:35',
      createdate: '2003-10-02 01:25:35',
      stars: '5'
    },
    {
      review_id: '7226',
      reviewbody:
        "I'm not the biggest fan of the Eames' work. For some odd reason, I find they're animated stories stiff and too formal. In the case of The Information Machine, it's a cartoon about computers told somewhat straight forward as the Eames can tell it. The animation isn't all THAT great, one thing that does save it is the score by Elmer Bernstein.",
      reviewtitle: 'Animation for scientists',
      reviewer: 'Spuzz',
      reviewdate: '2003-11-07 16:42:10',
      createdate: '2003-11-07 16:42:10',
      stars: '3'
    }
  ];
  const tree = renderer.create(<ReviewsSection reviews={reviews} />).toJSON();
  expect(tree).toMatchSnapshot();
});
