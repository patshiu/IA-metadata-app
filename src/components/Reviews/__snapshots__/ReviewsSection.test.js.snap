// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders review section with no reviews correctly 1`] = `
<div
  className="row"
>
  <div
    className="section item-reviews"
  >
    <h2>
      Reviews
       
    </h2>
    <p>
      This item has no reviews.
    </p>
  </div>
</div>
`;

exports[`renders review section with several reviews correctly 1`] = `
<div
  className="row"
>
  <div
    className="section item-reviews"
  >
    <h2>
      Reviews
       
      (3)
    </h2>
    <div
      className="review-item"
    >
      <h3>
        If I only had a heart
      </h3>
      <div>
        <div
          aria-description="Rating: 3 of 5 stars"
        >
          ★★★☆☆
        </div>
         by 
        dynayellow
         on 
        2003-09-15 15:54:20
      </div>
      <p>
        Wonderfully trippy little cartoon on the function of computers. The color's a little muddy, but it explains the use of computers--in the service of man, not its master--as a tool for the control of information and work, design, and for simulating reality.

Not sure what need this short really fulfills, but it's fun (kinda like a Fractured Fairytale-style), doesn't pander, and doesn't panic.
      </p>
    </div>
    <div
      className="review-item"
    >
      <h3>
        A gem from the Late 50s - Charles and Rae Eames
      </h3>
      <div>
        <div
          aria-description="Rating: 5 of 5 stars"
        >
          ★★★★★
        </div>
         by 
        Steve R
         on 
        2003-10-02 01:25:35
      </div>
      <p>
        This short film was written, produced and directed by Charles and Rae Eames for the IBM Pavillion at the 1958 Brussels World's Fair. Animation by John Whitney. Music by Elmer Bernstein. Topic is the computer in the context of human development.  MUST VIEW!
      </p>
    </div>
    <div
      className="review-item"
    >
      <h3>
        Animation for scientists
      </h3>
      <div>
        <div
          aria-description="Rating: 3 of 5 stars"
        >
          ★★★☆☆
        </div>
         by 
        Spuzz
         on 
        2003-11-07 16:42:10
      </div>
      <p>
        I'm not the biggest fan of the Eames' work. For some odd reason, I find they're animated stories stiff and too formal. In the case of The Information Machine, it's a cartoon about computers told somewhat straight forward as the Eames can tell it. The animation isn't all THAT great, one thing that does save it is the score by Elmer Bernstein.
      </p>
    </div>
  </div>
</div>
`;

exports[`renders review section with undefined reviews object correctly 1`] = `
<div
  className="row"
>
  <div
    className="section item-reviews"
  >
    <h2>
      Reviews
       
    </h2>
    <p>
      This item has no reviews.
    </p>
  </div>
</div>
`;
